app/assets/src/components/views/SampleViewV2/ReportTableV2.jsx

```diff
@@ -0,0 +1,404 @@
+        dataKey: "ntnrSelector",
+        disableSort: true,
+        headerClassName: cs.ntnrSelectorHeader,
+        headerRenderer: this.renderNtNrSelector,
+        width: 40,
+      },
+    ];
+  }
+
+  nestedSortFunction = ({ data, path, sortDirection, nullValue, limits }) => {
+    // Uses lodash's orderBy function.
+    // It uses a sorting key that enables nested sorting of genus and species, while guaranteeing that
+    // genus is always on top of its children species
+    return orderBy(
+      [
+        rowData =>
+          rowData.genus
+            ? getOr(nullValue, ["genus"].concat(path), rowData)
+            : getOr(nullValue, path, rowData),
+        // this value guarantees that we keep species within their genus, even if the first value is duplicated
+        rowData => (rowData.genus ? rowData.genus.taxId : rowData.taxId),
+        rowData =>
+          rowData.genus
+            ? getOr(nullValue, path, rowData)
+            : sortDirection === "asc"
+              ? limits[0]
+              : limits[1],
+      ],
+      [sortDirection, sortDirection, sortDirection],
+      data
+    );
+  };
+
+  nestedNtNrSortFunction = ({ path, ...props }) => {
+    const { dbType } = this.state;
+    return this.nestedSortFunction({ path: [dbType].concat(path), ...props });
+  };
+
+  toggleExpandGenus = ({ taxId }) => {
+    const { expandedGenusIds } = this.state;
+    expandedGenusIds.delete(taxId) || expandedGenusIds.add(taxId);
+
+    this.setState({ expandedGenusIds: new Set(expandedGenusIds) });
+  };
+
+  renderAggregateScore = ({ cellData, rowData }) => {
+    return (
+      <div className={cs.annotatedData}>
+        <div className={cs.icon}>{rowData.highlighted && <InsightIcon />}</div>
+        <div className={cs.data}>
+          {TableRenderers.formatNumberWithCommas(Number(cellData).toFixed(0))}
+        </div>
+      </div>
+    );
+  };
+
+  renderName = ({ cellData, rowData }) => {
+    return (
+      rowData && (
+        <React.Fragment>
+          <div className={cs.expandIcon}>
+            {rowData.taxLevel == "genus" ? (
+              <i
+                className={cx("fa", "fa-angle-right")}
+                onClick={withAnalytics(
+                  () => this.toggleExpandGenus({ taxId: rowData.taxId }),
+                  "PipelineSampleReport_expand-genus_clicked",
+                  { tax_id: rowData.taxId }
+                )}
+              />
+            ) : (
+              ""
+            )}
+          </div>
+          <div className={cx(rowData.taxLevel == "species" && cs.speciesName)}>
+            {cellData}
+          </div>
+        </React.Fragment>
+      )
+    );
+  };
+
+  renderNtNrDecimalValues = ({ cellData, decimalPlaces }) => {
+    return this.renderNtNrValues({
+      cellData: cellData.map(val =>
+        TableRenderers.formatNumberWithCommas(
+          Number(val).toFixed(decimalPlaces || 0)
+        )
+      ),
+    });
+  };
+
+  handleNtNrChange = selectedDbType => {
```

Comment:
Nit: Could we group all the render functions at the bottom and group similar functions together, instead of interspersing render functions with event handlers?

Reply:
Sure.